.travis.yml

```diff
@@ -24,6 +28,10 @@ matrix:
   - ghc: 8.6.5
     env: STACK_YAML="$TRAVIS_BUILD_DIR/stack.yaml" DEPLOY=yes
 
+  - language: nix
+    install: echo 'Running nix...'
+    script: nix-shell --run 'cabal new-test all --enable-tests'
```

Comment:
Probably the `before-script` part should be written under this `matrix` item near `script`. Also, I would prefer uniform style, like this:

```yaml
    install:
      - ...
    before_script:
      - ...
      - ...
    script:
      - ...
```